Task 1: Read the CSV file in Pandas and create a DataFrame named Grc_df. What is the number of
rows and columns in Grc_df? Print the first 10 and last 10 rows of Grc_df. 


In [119]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from scipy import stats
import seaborn as sns

url = "Grocery_dataset.csv"

Grc_df = pd.read_csv(url)

rows, cols = Grc_df.shape

print(f"Rows {rows}, Columns: {cols} \n")
print(Grc_df.head(10))
print(Grc_df.tail(10))

Rows 5000, Columns: 12 

  Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0           FDA15        9.300          Low Fat         0.016047   
1           DRC01        5.920          Regular         0.019278   
2           FDN15       17.500          Low Fat         0.016760   
3           FDX07       19.200          Regular         0.000000   
4           NCD19        8.930          Low Fat         0.000000   
5           FDP36       10.395          Regular         0.000000   
6           FDO10       13.650          Regular         0.012741   
7           FDP10          NaN          Low Fat         0.127470   
8           FDH17       16.200          Regular         0.016687   
9           FDU28       19.200          Regular         0.094450   

               Item_Type  Item_MRP Outlet_Identifier  \
0                  Dairy  249.8092            OUT049   
1            Soft Drinks   48.2692            OUT018   
2                   Meat  141.6180            OUT049   
3 

Task 2: Are there any null values in the Grc_df? If yes, then in which columns and how many?
Finally, handle these null values using any strategy shown during the labs. 


In [120]:

Grc_df = Grc_df.replace(r'^\s*$', np.nan, regex=True)

print(Grc_df.isnull().sum())




Item_Identifier                 0
Item_Weight                   818
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  1439
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64


We have 818 null values in the Item weight column and 1439 null values in the Outlet size column. 

In [121]:

display(Grc_df['Item_Identifier'].unique())

Grc_df.groupby('Item_Identifier')['Item_Weight'].nunique()

    

<StringArray>
['FDA15', 'DRC01', 'FDN15', 'FDX07', 'NCD19', 'FDP36', 'FDO10', 'FDP10',
 'FDH17', 'FDU28',
 ...
 'FDP21', 'FDD33', 'DRC49', 'FDB10', 'FDS39', 'NCU29', 'NCP14', 'FDC48',
 'FDW14', 'FDS36']
Length: 1538, dtype: str

Item_Identifier
DRA12    1
DRA24    1
DRA59    1
DRB01    1
DRB13    1
        ..
NCZ30    1
NCZ41    1
NCZ42    1
NCZ53    1
NCZ54    1
Name: Item_Weight, Length: 1538, dtype: int64

We can see that all items under the same Item_Identifier have the same weight. With this we can assume that this is true for the items with null values 

In [122]:
Grc_df['Item_Weight'] = Grc_df.groupby('Item_Identifier')['Item_Weight'].transform(lambda x: x.fillna(x.mean()))

Grc_df.isnull().sum()

Item_Identifier                 0
Item_Weight                    35
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  1439
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

We have sucessfully filled most item weight by using the group average/mean values, however there asre still 35 vlaues that coul dnot be filled and we will drop these.

In [123]:
Grc_df = Grc_df.dropna(subset = ['Item_Weight'])

Grc_df.isnull().sum()

Item_Identifier                 0
Item_Weight                     0
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  1439
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

In [124]:
Grc_df.groupby('Outlet_Type')['Outlet_Size'].value_counts(dropna=False)


Outlet_Type        Outlet_Size
Grocery Store      NaN             328
                   Small           270
Supermarket Type1  Small          1118
                   NaN            1111
                   High            558
                   Medium          541
Supermarket Type2  Medium          526
Supermarket Type3  Medium          513
Name: count, dtype: int64

Task 3: How many unique Outlet Sizes are there in the Grc_df? Which outlet size is maximum, and
which is minimum? 

In [125]:
Grc_df['Outlet_Size'].unique()

<StringArray>
['Medium', nan, 'High', 'Small']
Length: 4, dtype: str

There are 3 unique sizes; Small, Medium and High. High is the maximum and small is the minimum.

Taskk 4: How many unique Item Fat Content types are in the Grc_df? List them. Do you see any
issues with the Item Fat Content types? If yes, then handle this issue. 


In [126]:
Grc_df['Item_Fat_Content'].unique()

<StringArray>
['Low Fat', 'Regular', 'low fat', 'LF', 'reg']
Length: 5, dtype: str

The column has inconsistent labels for the same categories. The values should be standarized to "Regular" and "Low Fat"

In [127]:
Grc_df['Item_Fat_Content'] = Grc_df['Item_Fat_Content'].replace({'LF': 'Low Fat', 'low fat': 'Low Fat', 'reg': 'Regular'})

Grc_df['Item_Fat_Content'].unique()

<StringArray>
['Low Fat', 'Regular']
Length: 2, dtype: str

Task 5: Drop the columns having index values of 0, 6 and create a new DataFrame Grc_new_df. 